gravitee-am-reporter/gravitee-am-reporter-kafka/src/main/java/io/gravitee/am/reporter/kafka/audit/KafkaAuditReporter.java

```diff
@@ -34,123 +35,133 @@
 import io.vertx.core.Vertx;
 import io.vertx.kafka.client.producer.KafkaProducer;
 import io.vertx.kafka.client.producer.KafkaProducerRecord;
-import org.apache.kafka.common.serialization.Serializer;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.producer.ProducerConfig;
 import org.apache.kafka.common.serialization.StringSerializer;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.InitializingBean;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.StringUtils;
 
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Properties;
 
 /**
  * @author Florent Amaridon
  * @author Visiativ
  */
+@Slf4j
 public class KafkaAuditReporter extends AbstractService<Reporter> implements AuditReporter, InitializingBean {
 
-  private static Logger LOGGER = LoggerFactory.getLogger(KafkaAuditReporter.class);
+    private static final String SCHEMA_REGISTRY_URL_KEY = "schema.registry.url";
 
-  @Autowired
-  private Vertx vertx;
+    @Autowired
+    private Vertx vertx;
 
-  @Autowired
-  private KafkaReporterConfiguration config;
+    private KafkaReporterConfiguration config;
 
-  @Autowired
-  private GraviteeContext context;
+    @Autowired
+    private GraviteeContext context;
 
-  @Autowired
-  private Node node;
+    @Autowired
+    private Node node;
 
-  private final DtoMapper dtoMapper;
+    private KafkaProducer<String, AuditMessageValueDto> producer;
 
-  private KafkaProducer<String, AuditMessageValueDto> producer;
+    public KafkaAuditReporter(KafkaReporterConfiguration config) {
+        this.config = config;
+    }
+
+    @Override
+    public boolean canHandle(Reportable reportable) {
+        return reportable instanceof Audit;
+    }
+
+    @Override
+    public void report(Reportable reportable) {
+        log.debug("Report({})", reportable);
+        if (producer != null) {
+            AuditMessageValueDto value = DtoMapper.map((Audit) reportable, this.context, this.node);
+            this.producer.write(
+                    KafkaProducerRecord.create(this.config.getTopic(), this.context.getDomainId(), value));
+        } else {
+            log.debug("Producer is null, ignore reportable");
+        }
+    }
+
+    @Override
+    public void afterPropertiesSet() {
+        if (context == null) {
+            context = GraviteeContext.defaultContext(null);
+        }
+    }
```

Comment:
If you replaced field-level `@Autowired` with constructor injection, we could remove this method as well.
Also isn't this dead code? I'm not sure it's actually possible for `@Autowired` field to be null **after** injection - won't Spring complain about unmatched dependency?